client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionDialog.tsx

```diff
@@ -1,76 +1,227 @@
+            saveInteractions(interactionDataToSave);
+        }
+    };
+    const onPlaceSubtypeChange = (newValue: PlaceSubType | null) => {
+        if (newValue) {
+            setPlaceSubtypeName(newValue?.displayName);
+            methods.setValue(InteractionEventDialogFields.PLACE_SUB_TYPE, newValue.id, { shouldValidate: true });
+        } else {
+            setPlaceSubtypeName('');
+            methods.setValue(InteractionEventDialogFields.PLACE_SUB_TYPE, null, { shouldValidate: true });
+        }
+    };
+
+    const validateAndHandleSubmit = methods.handleSubmit(
+        () => {
+            const filTimeValidationMessage = 'יש למלא שעה';
+            if (!isUnknownTime) {
+                if (!interactionStartTime) {
+                    methods.setError(InteractionEventDialogFields.START_TIME, {
+                        type: 'manual',
+                        message: filTimeValidationMessage
+                    });
                 }
+                if (!interactionEndTime) {
+                    methods.setError(InteractionEventDialogFields.END_TIME, {
+                        type: 'manual',
+                        message: filTimeValidationMessage
+                    });
+                }
+            }
+            if (Boolean(interactionStartTime && interactionEndTime) || isUnknownTime) {
+                onSubmit(methods.getValues())
+            }
+        })
+
+    return (
+        <FormProvider {...methods}>
+            <Dialog classes={{ paper: classes.dialogPaper }} open={isOpen} maxWidth={false}>
+                <DialogTitle className={classes.dialogTitleWrapper}>
+                    {dialogTitle}
+                </DialogTitle>
+                <DialogContent>
+
+                    <form id='interactionEventForm' onSubmit={validateAndHandleSubmit}>
+                        <InteractionEventForm
+                            isVisible={!isAddingContacts}
+                            interactionData={interactionData}
+                            isNewInteraction={isNewInteraction}
+                            onPlaceSubTypeChange={onPlaceSubtypeChange}
+                        />
+                        <ContactsTabs isVisible={isAddingContacts} />
+                    </form>
+                </DialogContent>
+                <DialogActions className={`${classes.dialogFooter}`}   >
+                    {
+                        <Tooltip title={isContinueToContactsEnable() ? '' : 'לא ניתן לעבור ליצירת מגעים מבלי להזין "סוג אתר/תת סוג'}>
+                            <div>
+                                <Button disabled={!isContinueToContactsEnable()} variant='text' className={classes.changeEventSubFormButton}
+                                    onClick={async () => {
+                                        await methods.trigger()
+                                        isContinueToContactsEnable() && setIsAddingContacts(!isAddingContacts)
```

Comment:
why is ```isContinueToContactsEnable()``` needed here, if it is disabled it won't go to onClick

Reply:
Once you enter to a new blank Contact  dialog, validation is not triggered (that's how it works..) therefore the button is not disabled. Once the user clicks on the button - validation is triggered and if has errors it becomes disabled - the same way form submit works!